sdk-development-guidelines.md

```diff
@@ -0,0 +1,200 @@
+* MAY use any language/framework/tool for other supporting code
+
+#### Objective-C Best Practices (iOS <img src="images/ios.png" alt="drawing" width="15"/>):
+
+* If the feature your module provides only works with more recent versions of iOS you MUST do proper feature-checking with e.g. `respondsToSelector:` and still target iOS 10.0 (or more recent version) for the deployment target.
+* Nullability: MUST use `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END` in header files, and explicitly add nullable when needed. The `_Nullable` style syntax is preferred to `__nullable`.
+* SHOULD NOT declare any static variable in public headers, as it can cause problems with Cocoapods.
+* SHOULD declare a public (and exported) constant and assign its value in a .m source file.
+* SHOULD use #pragma mark to logically separate the source code
+* [Symbol visibility](https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/DynamicLibraryDesignGuidelines.html#//apple_ref/doc/uid/TP40002013-SW18): All classes (public and private) and public constants SHOULD be exported (UIKIT_EXTERN or equivalent). When code is built with `-fvisibility=hidden`, classes not exported by dynamic frameworks don't exist in the ObjC runtime, which is why it's needed on private classes as well. It may be an Apple bug.
+    * Note: You do not have to export protocols and categories, notwithstanding their visibility.
+* MAY embed private classes in the .m file that uses them, instead of creating a separate .h/.m pair.
+* If a value is only ever used once, it's ok to not define a constant for it as long as the intent is clear (e.g. values passed to `+[UColor colorWithRed:green:blue:alpha:])` or documented with a comment.
+
+### Coding Style
+
+* Code MUST follow a style guide such as:
+    * [Kotlin's official coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html)
+        * To configure the IntelliJ formatter according to this style guide, please install Kotlin plugin version 1.5.0 or newer, go to Settings | Editor | Code Style | Kotlin, click on "Set from…" link in the upper right corner, and select "Predefined style / Kotlin style guide" from the menu
+    * [Google Java Style](https://google.github.io/styleguide/javaguide.html)
+    * [Swift](https://github.com/raywenderlich/swift-style-guide)
+    * [Objective-C](https://github.com/raywenderlich/objective-c-style-guide)
+* Existing modules SHOULD follow a coding style guide. Always strive for **consistency** within a project's codebase
+* SHOULD use static code analysis, such as [detekt](https://github.com/arturbosch/detekt), clang-format, [SwiftLint](https://github.com/realm/SwiftLint), or oclint to minimise errors and minimise debates and formatting fix-up commits in PRs. It is RECOMMENDED to integrate these tools with the CI workflow as pre-commit git hooks.
```

Comment:
Please remove clang-format and oclint from the list

Reply:
removed